sql-plugin/src/main/spark322/scala/com/nvidia/spark/rapids/shims/spark322/ScanExecShims.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2022-2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*** spark-rapids-shim-json-lines
+{"spark": "322"}
+spark-rapids-shim-json-lines ***/
+package com.nvidia.spark.rapids.shims
+
+import com.nvidia.spark.rapids._
+
+import org.apache.spark.sql.execution.{FileSourceScanExec, SparkPlan}
+import org.apache.spark.sql.execution.datasources.v2.BatchScanExec
+import org.apache.spark.sql.rapids.GpuFileSourceScanExec
+
+object ScanExecShims {
+  def tagGpuFileSourceScanExecSupport(meta: SparkPlanMeta[FileSourceScanExec]): Unit =
+    GpuFileSourceScanExec.tagSupport(meta)
+
+  def execs: Map[Class[_ <: SparkPlan], ExecRule[_ <: SparkPlan]] = Seq(
+    GpuOverrides.exec[FileSourceScanExec](
+      "Reading data from files, often from Hive tables",
+      ExecChecks(
+        (TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.STRUCT + TypeSig.MAP +
+          TypeSig.ARRAY + TypeSig.BINARY + TypeSig.DECIMAL_128).nested(),
+        TypeSig.all),
+      (fsse, conf, p, r) => {
+        // TODO: HybridScan supports DataSourceV2
+        // TODO: HybridScan only supports Spark 32X for now.
```

Comment:
Is this 3.2.2 or all 3.2.X?

Reply:
Now:
- supports all Spark versions 32x,33x,34x,35x, CDH, Databricks.
- throw error if runtime is CDH or Databricks

#### Background
Gluten doc shows only supports 322,331,342,351,  but actually:
- Gluten bundle 322 jar supports: Spark 320,321,322,323,324,325
- Gluten bundle 331 jar supports Spark 33x
- Gluten bundle 342 jar supports Spark 34x
- Gluten bundle 351 jar supports Spark 35x

#### version checks
In Hybrid repo, we checks the Gluten bundle major and minor versions are the same as  Spark major and minor versions.  This means: Gluten bundle 3.2.2 version satisfies Spark 3.2.0, 3.2.1, 3.2.2, 3.2.3, 3.2.4, 3.2.5. 
Spark version is in this pattern: x (Major).y(Minor).z(Fix)

Please refer to ScanExecShims in 320 folder:
```
/*** spark-rapids-shim-json-lines
{"spark": "320"}
{"spark": "321"}
{"spark": "321cdh"}
{"spark": "322"}
{"spark": "323"}
{"spark": "324"}
spark-rapids-shim-json-lines **
```
Please refer to ScanExecShims in 330 folder
```
/*** spark-rapids-shim-json-lines
{"spark": "330"}
{"spark": "330cdh"}
{"spark": "330db"}
{"spark": "331"}
{"spark": "332"}
{"spark": "332cdh"}
{"spark": "332db"}
{"spark": "333"}
{"spark": "334"}
{"spark": "340"}
{"spark": "341"}
{"spark": "341db"}
{"spark": "342"}
{"spark": "343"}
{"spark": "344"}
{"spark": "350"}
{"spark": "350db143"}
{"spark": "351"}
{"spark": "352"}
{"spark": "353"}
{"spark": "400"}
spark-rapids-shim-json-lines ***/
```